python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -349,15 +344,31 @@ def reset(self, errors_fatal=False):
             if not self.provider:
                 self.provider = _get_node_provider(self.config["provider"],
                                                    self.config["cluster_name"])
-            # Check whether we can enable the resource demand scheduler.
-            if "available_node_types" in self.config:
-                self.available_node_types = self.config["available_node_types"]
-                self.resource_demand_scheduler = ResourceDemandScheduler(
+
+            self.available_node_types = self.config["available_node_types"]
+            upscaling_speed = self.config.get("upscaling_speed")
+            aggressive = self.config.get("autoscaling_mode") == "aggressive"
+            target_utilization_fraction = self.config.get(
+                "target_utilization_fraction")
+            if upscaling_speed:
+                upscaling_speed = float(upscaling_speed)
+            elif aggressive:
+                upscaling_speed = float(self.config["max_workers"])
+            elif target_utilization_fraction:
+                upscaling_speed = 1 / max(target_utilization_fraction, 0.001)
+            else:
+                upscaling_speed = 1.0
+            if self.resource_demand_scheduler:
```

Comment:
We shouldn't do this since you can change the instance type of any node type

Reply:
I am confused, we call reset_config which handles it, can you please double check?

Reply:
But you are doing that to "make sure inferered resources are not lost."

What I'm saying is that you _want_ to lose inferred resources to make sure you don't have wrong resources.

Reply:
But I already do that in the case you mention, I overwrite if the node config changes, checkout reset_config function.